File: FrontendGestionDesTickets/src/app/models/ticket.ts
import {User} from './user';

export interface Ticket {
  id?: number;
  titre: string;
  description: string;
  statut: 'OUVERT' | 'EN_COURS' | 'RESOLU' | 'FERME';
  priorite: 'BASSE' | 'MOYENNE' | 'HAUTE';
  createur: User;
  technicien?: User;
  dateCreation: string;
  dateMiseAJour?: string;
}

File: src/main/java/org/ben/gestion_des_tickets/models/Ticket.java
package org.ben.gestion_des_tickets.models;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Entity
@Table(name = "tickets")
@Data
public class Ticket {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String titre;

    private String description;

    @Enumerated(EnumType.STRING)
    private Statut statut;

    @Enumerated(EnumType.STRING)
    private Priorite priorite;

    private LocalDateTime dateCreation;

    private LocalDateTime dateMiseAJour;

    @ManyToOne
    @JoinColumn(name = "id_employe")
    private User employe;

    @ManyToOne
    @JoinColumn(name = "id_technicien", nullable = true)
    private User technicien;

    public enum Statut {
        OUVERT, EN_COURS, RESOLU, FERME
    }

    public enum Priorite {
        FAIBLE, MOYENNE, ELEVEE, CRITIQUE
    }
}
